core/dbt/task/run.py

```diff
@@ -683,136 +611,241 @@ def _is_incremental(self, model) -> bool:
+                # or not. Thus, lets get a copy of the results as they exist right "now".
+                frozen_batch_results = deepcopy(batch_results)
+                self.merge_batch_results(result, frozen_batch_results)
+                self._update_result_with_unfinished_batches(result, batches)
+                return result
+
+            # breifly sleep so that this thread doesn't go brrrrr while waiting
+            time.sleep(0.1)
 
         # Only run "last" batch if there is more than one batch
         if len(batches) != 1:
             # Final batch runs once all others complete to ensure post_hook runs at the end
-            self._submit_batch(
-                node=node,
-                adapter=runner.adapter,
+            self.parent_task._submit_batch(
+                node=model,
+                adapter=self.adapter,
                 relation_exists=relation_exists,
                 batches=batches,
                 batch_idx=batch_idx,
                 batch_results=batch_results,
-                pool=pool,
+                pool=self.pool,
                 force_sequential_run=True,
                 skip=skip_batches,
             )
 
         # Finalize run: merge results, track model run, and print final result line
-        runner.merge_batch_results(result, batch_results)
-        track_model_run(runner.node_index, runner.num_nodes, result, adapter=runner.adapter)
-        runner.print_result_line(result)
+        self.merge_batch_results(result, batch_results)
 
         return result
 
+
+class RunTask(CompileTask):
+    def __init__(
+        self,
+        args: Flags,
+        config: RuntimeConfig,
+        manifest: Manifest,
+        batch_map: Optional[Dict[str, BatchResults]] = None,
+    ) -> None:
+        super().__init__(args, config, manifest)
+        self.batch_map = batch_map
+
+    def raise_on_first_error(self) -> bool:
+        return False
+
+    def get_hook_sql(self, adapter, hook, idx, num_hooks, extra_context) -> str:
+        if self.manifest is None:
+            raise DbtInternalError("compile_node called before manifest was loaded")
+
+        compiled = self.compiler.compile_node(hook, self.manifest, extra_context)
+        statement = compiled.compiled_code
+        hook_index = hook.index or num_hooks
+        hook_obj = get_hook(statement, index=hook_index)
+        return hook_obj.sql or ""
+
+    def handle_job_queue(self, pool, callback):
+        node = self.job_queue.get()
+        self._raise_set_error()
+        runner = self.get_runner(node)
+        # we finally know what we're running! Make sure we haven't decided
+        # to skip it due to upstream failures
+        if runner.node.unique_id in self._skipped_children:
+            cause = self._skipped_children.pop(runner.node.unique_id)
+            runner.do_skip(cause=cause)
+
+        if isinstance(runner, MicrobatchModelRunner):
+            runner._parent_task = self
+            runner._pool = pool
```

Comment:
nit: these should probably go through setters instead of updating the 'private' attrs directly.

Reply:
Fixed in [a32a7ae](https://github.com/dbt-labs/dbt-core/pull/11332/commits/a32a7ae553b8a8a19051c39e90dbfb031d8da82c) 🙂